src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -95,7 +95,7 @@ public async Task<ConnectNewHeadersResult> ConsensusTipChangedAsync()
 
             using (await this.asyncLock.LockAsync().ConfigureAwait(false))
             {
-                if (this.cachedHeaders.Count != 0)
+                if (this.cachedHeaders.Count != 0 && this.CanConsumeCache())
```

Comment:
Should we not disable this if `count == 1`?

Reply:
No. I think this case is handled before header even makes it to the cashe